Automattic/wp-calypso
# Virtual themes: Open site assembler

Fixes https://github.com/Automattic/wp-calypso/issues/75332

## Proposed Changes

Open the site assembler after selecting a virtual theme in the design picker so users can further customize the assembled design if the following conditions are satisfied:
- Viewport is large ([the assembler step is disabled otherwise](https://github.com/Automattic/wp-calypso/blob/25dfa286b5e056dac78e11e6b537e4122a0fb6de/packages/design-picker/src/components/pattern-assembler-cta/index.tsx#L26-L28)).
- The `pattern-assembler/dotcompatterns` feature flag is enabled ([patterns used by virtual themes are unavailable otherwise](https://github.com/Automattic/wp-calypso/blob/25dfa286b5e056dac78e11e6b537e4122a0fb6de/client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/patterns-data.ts#L593-L595)).

https://user-images.githubusercontent.com/1233880/233394594-460e263c-40dd-49fa-9cba-637ee5ef72ee.mov

## Testing Instructions

- Use the Calypso live link below.
- Go to `/setup/site-setup/designSetup?siteSlug=<SITE_DOMAIN>`.
- Select a virtual theme and continue with the design selection.
- Make sure you land in the site assembler.
- Make sure the pattern of the virtual theme is automatically pre-selected.
- Make sure you can add/remove/reorder patterns.
- Complete the site setup.
- Verify you get a site with BC3 and a homepage template with the expected patterns.
- Go back to `/setup/site-setup/designSetup?siteSlug=<SITE_DOMAIN>`.
- Resize the browser window to simulate a mobile viewport.
- Make sure you don't land in the site assembler after selecting a virtual theme.
